x-pack/plugin/search-business-rules/src/main/java/org/elasticsearch/xpack/searchbusinessrules/PinnedQueryBuilder.java

```diff
@@ -53,8 +53,8 @@ public class PinnedQueryBuilder extends AbstractQueryBuilder<PinnedQueryBuilder>
     public static final String NAME = "pinned";
     public static final int MAX_NUM_PINNED_HITS = 100;
```

Comment:
This is interesting - the pinned query only allows 100 pinned documents, if there are more it returns an error.
For the rules query, if there are more than pinned this limit, than maybe we should trim the list of pinned docs to this limit.

Reply:
I like that idea, thanks. Question, would it be better to trim here, or to return an error indicating that query rules need to be updated?

Reply:
Per discussion we will trim this and confirm with @aznick